test/web/test_server.py

```diff
@@ -69,3 +77,66 @@ def test_parse_address(source, result):
 
     else:
         assert parse_address(source) == result
+
+
+def test_custom_wsgi_app_is_used(container_factory, web_config):
+    wsgi_app = Mock()
+
+    class CustomWebServer(WebServer):
+        def get_wsgi_app(self):
+            return wsgi_app
+
+    class CustomHttpRequestHandler(HttpRequestHandler):
+        server = CustomWebServer()
+
+    http = CustomHttpRequestHandler.decorator
+
+    class CustomServerExampleService(object):
+        name = 'customserverservice'
+
+        @http('GET', '/')
+        def do_index(self, request):
+            return ''
+
+    container = container_factory(CustomServerExampleService, web_config)
+    with patch.object(CustomWebServer, 'get_wsgi_server') as get_wsgi_server:
+        container.start()
+
+    get_wsgi_server.assert_called_once_with(ANY, wsgi_app)
+
+
+def test_custom_wsgi_server_is_used(
+    container_factory, web_config, web_config_port, web_session
+):
+    def custom_wsgi_app(environ, start_response):
+        start_response('200 OK', [])
+        return 'Override'
```

Comment:
This is perfect. Demonstrates how to use the feature as well as testing that it works 💯